src/mono/wasi/testassets/Http.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Runtime.CompilerServices;
+
+// keep in sync with src\mono\sample\wasi\http-p2\Program.cs
+public static class WasiMainWrapper
+{
+    public static async Task<int> MainAsync(string[] args)
+    {
+        Console.WriteLine("Hello, Wasi Console!");
+        for (int i = 0; i < args.Length; i ++)
+            Console.WriteLine($"args[{i}] = {args[i]}");
+
+        using HttpClient client = new();
+        client.Timeout = Timeout.InfiniteTimeSpan;
+        client.DefaultRequestHeaders.Accept.Clear();
+        client.DefaultRequestHeaders.Accept.Add(
+            new MediaTypeWithQualityHeaderValue("application/vnd.github.v3+json"));
+        client.DefaultRequestHeaders.Add("User-Agent", ".NET WASI unit test");
+        
+        var query="https://api.github.com/orgs/dotnet/repos?per_page=1";
```

Comment:
I'm worried about rate limits etc.
Could you intead spin an HTTP server from the WBT test?

Reply:
yeah this won't work. you can use our echo server instead: https://corefx-net-http11.azurewebsites.net/Echo.ashx

either hardcoded or include src/libraries/Common/tests/System/Net/Configuration.cs and src/libraries/Common/tests/System/Net/Configuration.Http.cs and use System.Net.Test.Common.Configuration.Http.RemoteEchoServer

Reply:
used azure echo instead